src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -2412,24 +2412,42 @@ private MetadataSupport()
 
             public override void ConvertToJavascript(JavascriptConversionContext context)
             {
-                var methodCallExpression = context.Node as MethodCallExpression;
-
-                if (methodCallExpression?.Method.Name != "Metadata"
-                    && methodCallExpression?.Method.Name != "GetMetadataFor")
+                if (!(context.Node is MethodCallExpression methodCallExpression))
                     return;
 
                 if (methodCallExpression.Method.DeclaringType != typeof(RavenQuery)
                     && methodCallExpression.Object?.Type != typeof(IAdvancedSessionOperations)
                     && methodCallExpression.Object?.Type != typeof(IAsyncAdvancedSessionOperations))
                     return;
 
-                context.PreventDefault();
-                var writer = context.GetWriter();
-                using (writer.Operation(methodCallExpression))
+                switch (methodCallExpression.Method.Name)
                 {
-                    writer.Write("getMetadata(");
-                    context.Visitor.Visit(methodCallExpression.Arguments[0]);
-                    writer.Write(")");
+                    case "LastModified":
+                    {
+                        context.PreventDefault();
+                        var writer = context.GetWriter();
+                        using (writer.Operation(methodCallExpression))
+                        {
+                            writer.Write("getMetadata(");
+                            context.Visitor.Visit(methodCallExpression.Arguments[0]);
+                            writer.Write(")['@last-modified']");
```

Comment:
can we use constant?

Reply:
```suggestion
                            writer.Write(")['" + Constants.Documents.Metadata.LastModified + "']");
```

This is what it will look like with a constant, the string version is better, I think

Reply:
I'm more concerned about finding all places with the usages. It got handy few times. And compiler should remove the concatenations and use const here IIRC?

Reply:
I can't think of any scenario where this would be something we'll need. But ... 🤷